apps/mobile/src/store/accounts/utils.ts

```diff
@@ -22,19 +22,34 @@ export const accountIcons = [
   'heart',
   'flag',
 ] as const;
-
 export type AccountIcon = (typeof accountIcons)[number];
-export type AccountStatus = 'active' | 'hidden';
 
-export const accountStoreSchema = z.object({
-  id: z.string(),
-  icon: z.string(),
-  name: z.string(),
-  status: z.string(),
-});
+const accountStatuses = ['active', 'hidden'] as const;
+export type AccountStatus = (typeof accountStatuses)[number];
+
 export interface AccountStore {
   id: string;
-  icon: AccountIcon;
-  name: string;
-  status: AccountStatus;
+  icon?: AccountIcon;
+  name?: string;
+  status?: AccountStatus;
+}
+
+export const accountStoreSchema = z.object({
+  id: z.string(),
+  icon: z.enum(accountIcons).optional(),
+  name: z.string().optional(),
+  status: z.enum(accountStatuses).optional(),
+}) satisfies z.ZodType<AccountStore>;
+
+export function deriveIconFromAccountId(accountId: string): AccountIcon {
+  let hash = 0;
+  for (let i = 0; i < accountId.length; i++) {
+    const char = accountId.charCodeAt(i);
+    hash = (hash << 5) - hash + char;
+    hash = hash & hash;
+  }
+  const index = Math.abs(hash) % accountIcons.length;
+  const icon = accountIcons[index];
```

Comment:
Maybe we can use @noble/hashes here?
```suggestion
  const index = new DataView(sha256(accountId).buffer).getUint32(0) % accountIcons.length;
  const icon = accountIcons[index];
```

Reply:
I thought about this as well, is a cleaner implementation, it's way slower though

```
 LOG  Hashing 1000 strings took 2.1094159483909607ms
 LOG  SHA256 Hashing 1000 strings took 284.6787919998169ms
```

Reply:
blake3 is faster but still ~170ms